Lib/test/test_tempfile.py

```diff
@@ -1000,6 +1016,21 @@ def test_del_on_close(self):
         finally:
             os.rmdir(dir)
 
+    def test_del_rolled_file(self):
+        # The rolled file should be deleted when the
+        # SpooledTemporaryFile object is deleted. This should raise a
+        # ResourceWarning since the file was not explicitly closed.
+        f = self.do_create(max_size=2)
+        f.write(b'foo')
+        filename = f.name
+        self.assertTrue(os.path.exists(filename))
+        with self.assertWarns(ResourceWarning):
```

Comment:
Do we also expect a `ResourceWarning` to be raised when the file is not rolled over? If yes, there should be a test for it. If not, then the `__del__` implementation can be removed.